client/src/partials/fiscal/fiscal.lock.js

```diff
@@ -85,10 +85,38 @@ angular.module('bhima.controllers')
       })
       .then(function (data7) {
         session.array7 = data7.data;
+        return getSolde(8, fy_id);
+      })
+      .then(function (data8) {
+        session.array8 = data8.data;
       })
       .then(function () {
+        var otherCharge = session.array8.filter(function (item) {
+          var accountNumber = String(item.account_number);
+          var accountIndex = accountNumber.substring(0,2);
+
+          return (accountIndex === '81' || accountIndex === '83' || accountIndex === '85' || accountIndex === '89');
+        });
+
+        var otherProfit = session.array8.filter(function (item) {
+          var accountNumber = String(item.account_number);
+          var accountIndex = accountNumber.substring(0,2);
+
+          return (accountIndex === '82' || accountIndex === '84' || accountIndex === '86' || accountIndex === '88');
```

Comment:
Using of modulo will be better, you convert the two first number into Number and you test if x % 2 is 0 (even number) or not (odd number)